### PR TITLE
Updates Ruby image to 2.7.6

### PR DESCRIPTION
**Pull request by ascoppa**

### CODE REVIEW

File: images/ruby/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,10 @@
-FROM circleci/ruby:2.7.3-node AS ruby
+FROM cimg/ruby:2.7.6-node AS ruby
 
 ENV WORKDIR /home/circleci
 WORKDIR $WORKDIR
-RUN mkdir tmp bin && sudo apt-get update
```
Comment (ldgnoack) [inline images/ruby/Dockerfile:5]: Is there any reason to split this line? May be debugging? It's recommended to use as few as possible RUN so you create less layers unless you need them.

If you are using `apt update` you can also add ` && rm -rf /var/lib/apt/lists/*` to clean apt cache and make your image smaller (not needed if you are based on official Debian or Ubuntu images, but I don't know what circleci uses as a base).

Comment (ascoppa) [reply images/ruby/Dockerfile:5]: >  It's recommended to use as few as possible RUN so you create less layers unless you need them.

Ohh I wasn't aware of that, I'll restore it back to one line.

File: images/ruby/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+RUN mkdir -p tmp bin && sudo apt-get update
```
Comment (ascoppa) [inline images/ruby/Dockerfile:5]: I got the following error after switching to the new image so I added the option `p` which tells mkdir to not create the directory if it already exists.

Screen Shot 2022-07-25 at 22 17 50

File: images/ruby/Dockerfile [hunk 3]
```diff
@@ -1,8 +1,8 @@
-FROM circleci/ruby:2.7.3-node AS ruby
+FROM cimg/ruby:2.7.6-node AS ruby
```
Comment (ascoppa) [inline images/ruby/Dockerfile:1]: > Moving from a legacy to next-gen image requires a change to the namespace. All legacy images have a Docker namespace of circleci, while next-gen images have a Docker namespace of cimg

https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034

**Comment by ascoppa** [comment]
@ldgnoack this is RFR again https://github.com/bespokepost/build_images/pull/6/commits/e91fbe8d8c489df7637895926ea298b1b26bda20 

Also I forgot to add the `IGNORE` label when I was building this PR, I was planning to add some extra comments to explain some of the changes before moving it to RFR. Anyway, they are available now.

**Review by ldgnoack — APPROVED** [review]
It looks good to me, there's some further enhancement that could be made but that would be some Dockerfile profiling sorcery that it's not needed at all.

BTW, last weekend I was reading about running old versions of ElasticSearch con ARM64 and found [this repo](https://github.com/dockhippie/elasticsearch) maintained by an experienced devops who implements many good practices.

**Pull request merged**